tests/integration/components/bs-button-test.js

```diff
@@ -105,21 +105,21 @@ module('Integration | Component | bs-button', function(hooks) {
 
   test('button text is changed according to button state', async function(assert) {
 
-    await render(hbs`{{bs-button defaultText="text1" loadingText="text2" textState=textState}}`);
+    await render(hbs`{{bs-button defaultText="text1" loadingText="text2" state=state}}`);
```

Comment:
The tests are not ideal IMHO, in that they use a non-public property (`state`/`textState`) for testing. Instead they should better test the real public API, which is not about setting `state`, but returning a promise and reacting to its state. 

But don't be offended, I am basically complaining about myself, who has introduced this in the first place! ;)

But this can be refactored in a separate PR as well...

Reply:
Was also confused about that one and therefor thought it's considered public API. Since there isn't any test that it actual reflects the state of returned promise, I will refactor in this PR.